### PR TITLE
Avoid a full search if using the default printer (2.1)

### LINKED ISSUE #931: Use Default Printer still searches for all printers when printing
You can skip searching for all printers by simply clicking "Use Default Printer", however when printing, it still searches all printers.

### PR DESCRIPTION
**Pull request by bberenz**
Fixes #931

### CODE REVIEW

File: src/qz/printer/info/WindowsPrinterMap.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
 public class WindowsPrinterMap extends NativePrinterMap {
     private static final Logger log = LogManager.getLogger(WindowsPrinterMap.class);
 
-    public synchronized NativePrinterMap putAll(PrintService[] services) {
+    public synchronized NativePrinterMap putAll(PrintService... services) {
```
Comment (tresf) [inline src/qz/printer/info/WindowsPrinterMap.java:15]: Why is this changed to vararg?

Comment (tresf) [reply src/qz/printer/info/WindowsPrinterMap.java:15]: Oh, vararg allows arrays and direct args, neat. Ok, disregard.

**Review by tresf — APPROVED** [review]

File: src/qz/printer/info/WindowsPrinterMap.java (hunk 1 above)
Comment (bberenz) [reply src/qz/printer/info/WindowsPrinterMap.java:15]: so PrintServiceMatcher can put just the default printer if nothing else has been searched for yet without having to create an array of one

Comment (tresf) [reply src/qz/printer/info/WindowsPrinterMap.java:15]: > so PrintServiceMatcher can put just the default printer if nothing else has been searched for yet without having to create an array of one

Thanks.  Yeah, after staring at it for a while, I realized it was a shortcut to allow one value or an array of values, which is a shortcut I never thought of using. :)

**Comment by klabarge** [comment]
This works. When the default printer is used, a full search is skipped.

### macOS
```
[DEBUG] 2022-03-11T13:18:53,795 @ qz.utils.PrintingUtilities:-1
	Using qz.printer.action.PrintPDF to print
[DEBUG] 2022-03-11T13:18:53,796 @ qz.printer.PrintServiceMatcher:-1
	Searching for PrintService matching PDFwriter
[DEBUG] 2022-03-11T13:18:53,858 @ qz.printer.PrintServiceMatcher:-1
	Matched default printer, skipping further search
```

### Windows
```
[DEBUG] 2022-03-11T13:50:49,790 @ qz.utils.PrintingUtilities:-1
        Using qz.printer.action.PrintPDF to print
[DEBUG] 2022-03-11T13:50:49,791 @ qz.printer.PrintServiceMatcher:-1
        Searching for PrintService matching PDFCreator
[DEBUG] 2022-03-11T13:50:49,792 @ qz.printer.PrintServiceMatcher:-1
        Matched default printer, skipping further search
```
